lara-typescript/src/lib/plugin-context.ts

```diff
@@ -30,12 +41,17 @@ export interface IPluginContext {
   classInfoUrl: string | null;
   /** URL to fetch a JWT. Includes generic `_FIREBASE_APP_` that should be replaced with app name. */
   firebaseJwtUrl: string;
-  /** Wrapped embeddable runtime context if plugin is wrapping some embeddable. */
-  wrappedEmbeddable: IEmbeddableContext | null;
+}
+
+export interface IPluginAuthoringContextOptions extends IPluginCommonOptions {
+  /** Type of the plugin */
+  type: "authoring";
+  /** URL used to save plugin authoring state. */
+  authorDataSaveUrl: string;
 }
 
 /** Data generated by LARA, passed to .initPlugin function, and then used to generate final IEmbeddableRuntimeContext */
-export interface IEmbeddableContext {
+export interface IEmbeddableContextOptions {
```

Comment:
Why is `Options` added to the name here?

Reply:
The original name `IEmbeddableContext` was confusing because it is different from the `IPluginRuntimeContext` that is part of the API and is passed to a plugin. Instead this is a set of options that is passed in order to generate the `IPluginRuntimeContext`.

Reply:
To me it is weird that we now have IPluginRuntimeContext and IPluginContextOptions, and IEmbeddableRuntimeContext and IEmbeddableContextOptions.

If we are going with Options then it seems better to have IPluginContext and IPluginContextOptions  or to have IPluginRuntimeContext and IPluginRuntimeContextOptions.  Removing Runtime and adding Options seems inconsistent.

I'm looking to see if there is a new IPluginAuthoringContext...

Reply:
Looking at the code it looks like IEmbeddableContext was also defined in the embeddable-runtime-context.ts file. It looks like that file was not changed by this PR.

Doh, I see there is a IPluginAuthoringContextOptions and IPluginRuntimeContextOptions.  Sorry. So I guess my only problem was really IEmbeddable* . I'm continuing to grok this to see if there is really anything that can be improved...

Reply:
Probably my rambling above is hiding the one thing that I think needs to change here:

IEmbeddableContext seems to also be defined in embeddable-runtime-context.ts and that file doesn't seem to be changed by this PR.  I don't know why it is defined twice, but if it is renamed it should be renamed there or the duplication should be fixed.